### PR TITLE
Upgrade to monolog/monolog 3.3.1

### PR DESCRIPTION
**Pull request by umutphp**
**Description**
While trying to use this lib with Laravel 10, the monolog/monolog package dependencies create a conflict. 

I just upgraded the monolog/monolog version.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "ext-json": "*",
     "ext-mbstring": "*",
     "ext-openssl": "*",
-    "monolog/monolog": "^1.16 || ^2.0"
+    "monolog/monolog": "^3.3"
```
Comment (pyrou) [inline composer.json:17]: Why dropping monolog 1 and 2 ?
And not simply allowing monolog 3 as following ?

```suggestion
    "monolog/monolog": "^1.16 || ^2.0 || ^3.0"
```

Comment (umutphp) [reply composer.json:17]: @pyrou 
You are right 👍🏻

**Comment by vadminn** [comment]
Dear @AlexandrosMor, @acampos1916, @peterojo, @rikterbeek, @Morerice, @michaelpaul, @candemiralp, @RokPopov!
Could you check this PR? his PR is the only blocker from updating the Laravel framework to the latest version for Adyen users. Updating the framework will improve the performance and security of our application and ensure compatibility with future updates.

I have noticed that there are three more PRs from the bots for the same library update.

I would appreciate it if you could take some time to review this PR and provide any feedback or suggestions. 

Thanks!

**Review by peterojo — APPROVED** [review]

**Comment by pyrou** [comment]
@umutphp Can you check the failing `composer validate --strict` from Github Action ?

A simple `composer update --lock` might resolve the issue.

**Comment by umutphp** [comment]
> @umutphp Can you check the failing `composer validate --strict` from Github Action ?
> 
> A simple `composer update --lock` might resolve the issue.

Yes, sure 👍🏻 .

**Comment by umutphp** [comment]
@pyrou `composer update --lock` is done. 

image

**Review by antolo-arch — DISMISSED** [review]

**Review by wboereboom — APPROVED** [review]

**Comment by pyrou** [comment]
According to new failing on Github Action, I would recommend to not update the `composer.lock` and completely revert it. 
What do you think about this ?

As I'm reading your PR description, you want Laravel 10 support. Me also. The primary goal of this PR is to "support" Monolog v3 not to strictly upgrade to Monolog v3. Am I right ?

**Comment by darthsoup** [comment]
> By doing this you can also add php 8.1 and 8.2 to your CI matrix, monolog V2 will naturally be installed on PHP 7.3 to PHP 8.0, and monolog v3 will be used on PHP 8.1 to PHP 8.2.

There is already a merge request for that at #442

**Comment by umutphp** [comment]
I agree with @pyrou  about the composer.lock file.

**Review by jillingk — APPROVED** [review]

**Comment by pyrou** [comment]
@umutphp Can we give a try with removed `composer.lock` ? GithubAction should be green

**Comment by jillingk** [comment]
> @umutphp Can we give a try with removed `composer.lock` ? GithubAction should be green

Hi @umutphp and @pyrou, we just discussed and agreed as well that we want to remove the composer.lock. If you can update the PR then we'll approve again and we can finally have this merged. Thanks a lot!

**Comment by umutphp** [comment]
@jillingk composer.lock file is removed 👍🏻

**Review by jillingk — APPROVED** [review]

**Pull request merged**